### DISCUSSION TITLE
中科大羽毛球场预约小程序脚本 - 逸風亭

### DISCUSSION THREAD

**Discussion by windshadow233** [windshadow233/BlogStorage#261] [Announcements]
https://blog.fyz666.xyz/blog/6435/

**Comment by hhr346** [comment]
现在的平台调用wx云函数来获取noise参数是不是无解了，这个云函数看了下开发者文档似乎是用户自己上传的自定义函数（
而且它这个128个字符的noise似乎是拼接成的，对相同的参数也有不同的结果。。。
就差这个参数了，我哭死

**Comment by windshadow233** [comment]
> @hhr346
> 现在的平台调用wx云函数来获取noise参数是不是无解了，这个云函数看了下开发者文档似乎是用户自己上传的自定义函数（
> 而且它这个128个字符的noise似乎是拼接成的，对相同的参数也有不同的结果。。。
> 就差这个参数了，我哭死

是的，这个我查了很多资料，应该是没法解决～可以考虑折中方法，手动把token抓来，然后一段时间内不会过期。或者用那种可视的自动化工具来搞。

**Comment by windshadow233** [comment]
> @hhr346
> 现在的平台调用wx云函数来获取noise参数是不是无解了，这个云函数看了下开发者文档似乎是用户自己上传的自定义函数（
> 而且它这个128个字符的noise似乎是拼接成的，对相同的参数也有不同的结果。。。
> 就差这个参数了，我哭死

顺便请教一下code的获取方法（

**Comment by hhr346** [comment]
> @windshadow233
> > @hhr346
> > 现在的平台调用wx云函数来获取noise参数是不是无解了，这个云函数看了下开发者文档似乎是用户自己上传的自定义函数（
> > 而且它这个128个字符的noise似乎是拼接成的，对相同的参数也有不同的结果。。。
> > 就差这个参数了，我哭死
> 
> 顺便请教一下code的获取方法（

现在的小程序提交订单的时候应该是还有个sign参数吧，观察是每个提交操作都会产生相应的sign。看了下网上教程把小程序反编译成js代码，然后在里面找到了sign的生成方法，大致就是对订单参数排序拼接处理之后加上一段字符串再取的md5码。

**Comment by windshadow233** [comment]
> @hhr346
> > @windshadow233
> > > @hhr346
> > > 现在的平台调用wx云函数来获取noise参数是不是无解了，这个云函数看了下开发者文档似乎是用户自己上传的自定义函数（
> > > 而且它这个128个字符的noise似乎是拼接成的，对相同的参数也有不同的结果。。。
> > > 就差这个参数了，我哭死
> > 
> > 顺便请教一下code的获取方法（
> 
> 现在的小程序提交订单的时候应该是还有个sign参数吧，观察是每个提交操作都会产生相应的sign。看了下网上教程把小程序反编译成js代码，然后在里面找到了sign的生成方法，大致就是对订单参数排序拼接处理之后加上一段字符串再取的md5码。

sign我倒是解决了，确实是通过反编译wxapkg找到的方法，我之前是解决不了code，然后手动抓token去预约，然后也是被noise参数搞的头大😅可能只能用按键精灵之类的东西去搞了
